src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -0,0 +1,270 @@
+"use client";
+
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMemo } from "react";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import {
+  Select,
+  SelectTrigger,
+  SelectContent,
+  SelectItem,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { User } from "@/types";
+import { MemberRole, UserStatus } from "@/lib/routes";
+
+const attendanceSchema = z
+  .object({
+    memberId: z.number({
+      required_error: "Please select user name",
+    }),
+    startTime: z.string().nonempty({ message: "Start time is required" }),
+    endTime: z.string().nonempty({ message: "End time is required" }),
+  })
+  .refine(
+    (data) => {
+      const startTime = new Date(`1970-01-01T${data.startTime}:00`);
```

Comment:
can we use her Date.now().replace for it

Reply:
Let's add a comment to explain what's going on here.